src/Validation/Validation.php

```diff
@@ -791,18 +791,24 @@ public static function email(mixed $check, ?bool $deep = false, ?string $regex =
     /**
      * Checks that the value is a valid backed enum instance or value.
      *
+     * Valid Options
+     *
+     *  - only => enum case or array of enum cases that are valid
+     *  - except => enum case or array of enum cases that are not valid
+     *
      * @param mixed $check Value to check
      * @param class-string<\BackedEnum> $enumClassName The valid backed enum class name
+     * @param array<string, mixed> $options
      * @return bool Success
      * @since 5.0.3
      */
-    public static function enum(mixed $check, string $enumClassName): bool
+    public static function enum(mixed $check, string $enumClassName, array $options = []): bool
```

Comment:
Changing method signature is a BC break.

Reply:
5.next then? As it is optional one?

Reply:
I think @markstory allows adding new defaulted parameters to public methods.

Reply:
Needs to be 5.next at a minimum as it would still break extension.

Reply:
The alternative would be 3 speaking methods:
- enum()
- enumOnly()
- enumExcept()

Reply:
I think `Validation` is a reasonable place to have more methods. With an already wide API like `Validation` more methods helps autocomplete users and documentation readers as they can ctrl-f and find all the methods at once. You can choose the behavior you want on method name alone without having to read the doc block.